src/repositories/__test__/qr.repo.integration.test.ts

```
@@ -0,0 +1,148 @@
+    it('QR 토큰을 생성하고 정상 조회할 수 있어야 한다', async () => {
+      const token = generateRandomToken();
+      const ip = '127.0.0.1';
+      const userAgent = 'test-agent';
+
+      await repo.createQRLoginToken(token, TEST_DATA.USER_ID, ip, userAgent);
+      const foundToken = await repo.findQRLoginToken(token);
+
+      expect(foundToken).not.toBeNull();
+      expect(foundToken?.token).toBe(token);
+      expect(foundToken?.is_used).toBe(false);
+      expect(new Date(foundToken!.expires_at).getTime()).toBeGreaterThan(new Date(foundToken!.created_at).getTime());
```

ESLint

[error] Forbidden non-null assertion.

(@typescript-eslint/no-non-null-assertion)